### PR TITLE
Fix matplotlib bug and assorted warnings

### LINKED ISSUE #52: Reports suddenly taking much longer to generate
# 🐛 Summary #

As of August 14, 2021, the trustymail reports are taking twice as long to run.  Here is a table showing some data I have collected:
| Date | Start | Finish | Time |
| --- | --- | --- | --- |
| 07/24 | 10:42 | 22:43 | ~12h |
| 07/31 | 10:47 | 23:24 | ~13h |
| 08/07 | 10:36 | 22:19 | ~12h |
| 08/14 | 10:47 | 11:38 (+1) | ~25h |
| 08/21 | 10:42 | 10:31 (+1) | ~24h |
| 08/28 | 10:38 | 12:54 (+1) | ~26h |

Of note:
- I have verified by inspecting the logs that it is the actual report generation that is taking longer, not the Elasticsearch queries for DMARC records.
- I have verified that the HTTPS reports are running in about the same amount of time.
- I recently rebuilt the `cisagov/trustymail_reporter` Docker image, so I am in the process of testing that.

## Expected behavior ##

I don't see anything that should have started making these reports take twice as long after August 7.

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request:
* Fixes the `import` of `matplotlib`.  Now that the code is running in an updated Docker image (after #51) the way we were doing it before caused an error.
* Updates some Python code that was using deprecated functions.
* Updates some LaTeX code to get rid of a warning.
* Modifies one usage of `\color` in a `longtable` LaTeX environment to use `\textcolor` instead in order to get rid of a new bug where table rows were twice as tall as necessary.
* Moves the configuration of the `background` package close to where it is actually used in order to get rid of a new bug where the DHS logo watermark was not appearing correctly.

## 💭 Motivation and context ##

This Docker image was not functioning as expected after merging #51, so the error fix in PR was necessary.  The warning fixes are just things I noticed when trying to debug #52.

Once it was working again, I had to fix the `longtable` and `background` issues described in the previous section in order to generate output that completely agreed with that produced this past weekend using a much older Docker image.

## 🧪 Testing ##

I tested by running the report generation in production with the new Docker image and comparing at the results to the reports generated over the weekend using a much older Docker image.  They appear identical to these eyes!

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

### CODE REVIEW

File: src/report/generate_trustymail_report.py [hunk 1]
```diff
@@ -251,9 +251,9 @@ def add_weak_crypto_data_to_domain(domain_doc, sslyze_data_all_domains):
         all_domains_cursor.close()
 
         # Get count of second-level domains an agency owns
-        self.__base_domain_count = self.__db.trustymail.find(
+        self.__base_domain_count = self.__db.trustymail.count_documents(
             {"latest": True, "agency.name": agency, "is_base_domain": True}
-        ).count()
+        )
```
Comment (dav3r) [inline src/report/generate_trustymail_report.py:256]: Nice performance improvement! 👍

**Review by dav3r — APPROVED** [review]
Excellent cleanup work and good job squashing those bugs! 🚫 🐞 👍

File: src/report/generate_trustymail_report.py (hunk 1 above)
Comment (jsf9k) [reply src/report/generate_trustymail_report.py:256]: I still don't understand how this suddenly became an issue after August 7, but I've wasted enough time trying to figure that out. :disappointed:

Comment (dav3r) [reply src/report/generate_trustymail_report.py:256]: Yeah, that's strange, but maybe that large influx of subdomains that we discussed pushed us over the edge somehow.  Anyway, thanks for finding something that works here.

File: src/report/trustymail_report.mustache [hunk 2]
```diff
@@ -344,6 +343,18 @@ opacity=0.04
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%% Table of Contents
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% DHS Image Watermark
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% The watermarks are known to me acting a little wonky.  I created
```
Comment (mcdonnnj) [inline src/report/trustymail_report.mustache]: I think this is what you meant?

```suggestion
% The watermarks are known to be acting a little wonky.  I created
```

**Review by mcdonnnj — APPROVED** [review]
Thanks for putting in the work on this 💪💪💪

**Pull request merged**